satellite/api.go

```diff
@@ -0,0 +1,526 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellite
+
+import (
+	"context"
+	"net"
+	"net/mail"
+	"net/smtp"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/internal/errs2"
+	"storj.io/storj/internal/post"
+	"storj.io/storj/internal/post/oauth2"
+	"storj.io/storj/internal/version"
+	"storj.io/storj/pkg/auth/grpcauth"
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/peertls/extensions"
+	"storj.io/storj/pkg/peertls/tlsopts"
+	"storj.io/storj/pkg/rpc"
+	"storj.io/storj/pkg/server"
+	"storj.io/storj/pkg/signing"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/accounting"
+	"storj.io/storj/satellite/accounting/live"
+	"storj.io/storj/satellite/accounting/rollup"
+	"storj.io/storj/satellite/console"
+	"storj.io/storj/satellite/console/consoleauth"
+	"storj.io/storj/satellite/console/consoleweb"
+	"storj.io/storj/satellite/contact"
+	"storj.io/storj/satellite/inspector"
+	"storj.io/storj/satellite/mailservice"
+	"storj.io/storj/satellite/mailservice/simulate"
+	"storj.io/storj/satellite/marketingweb"
+	"storj.io/storj/satellite/metainfo"
+	"storj.io/storj/satellite/nodestats"
+	"storj.io/storj/satellite/orders"
+	"storj.io/storj/satellite/overlay"
+	"storj.io/storj/satellite/repair/irreparable"
+	"storj.io/storj/satellite/repair/repairer"
+	"storj.io/storj/satellite/vouchers"
+)
+
+// APIConfig is the global config for the satellite API process
+type APIConfig struct {
+	Identity identity.Config
+	Server   server.Config
+	Version  version.Config
+
+	Contact contact.Config
+	Overlay overlay.Config
```

Comment:
might it be worth keeping `Discovery` in here until we remove it from satellite/peer too? or does that make things more complicated

Reply:
i took it out since it had no public/private endpoints nor was a dependency of anything else here.  Is it doing something that the API would need?

Reply:
never mind, i was trying to delete this earlier comment from my review and somehow managed to send it.

it makes sense to leave discovery out.